docs/reference/mapping/fields/synthetic-source.asciidoc

```diff
@@ -155,18 +198,91 @@ that ordering.
 
 [[synthetic-source-modifications-ranges]]
 ====== Representation of ranges
-Range field values (e.g. `long_range`) are always represented as inclusive on both sides with bounds adjusted accordingly. See <<range-synthetic-source-inclusive, examples>>.
+Range field values (e.g. `long_range`) are always represented as inclusive on both sides with bounds adjusted
+accordingly. See <<range-synthetic-source-inclusive, examples>>.
 
 [[synthetic-source-precision-loss-for-point-types]]
 ====== Reduced precision of `geo_point` values
-Values of `geo_point` fields are represented in synthetic `_source` with reduced precision. See <<geo-point-synthetic-source, examples>>.
+Values of `geo_point` fields are represented in synthetic `_source` with reduced precision. See
+<<geo-point-synthetic-source, examples>>.
+
+[[synthetic-source-keep]]
+====== Keeping the origical source
+
+It is possible to record the original source of an object or field, at extra storage cost, using param
+`synthetic_source_keep`. The default value is `none`; setting it to `arrays` leads to storing the original source for
```

Comment:
Maybe we shouldn't guarantee that we store original source when `synthetic_source_keep` is set to `arrays` is enabled?  Maybe we we should just guarantee the array ordering is guaranteed? This gives us better ways to reduce the overhead later on without breaking bwc.

Reply:
If we need to main bwc with things like field: [1, 2, [3], [[4, [5]]], ["5"], 6] , then that makes things more difficult. Ideally we would synthesize this as: field: [1, 2, 3, 5, 5, 6] (assuming this is a number field).

Reply:
But this is how it's implemented right now. Are you suggesting that we don't document the current behavior? I think it'll be a breaking change if we change it to return ` [1, 2, 3, 5, 5, 6]` in the example above, whether we document it or not..

Reply:
I guess we can be explicit that we don't guarantee the exact form.. is this what you have in mind?

Reply:
> I guess we can be explicit that we don't guarantee the exact form.. is this what you have in mind?

Apologies, yes, this is what I mean.

Reply:
I've updated the wording to reflect that this may change, ptal.